lib/lhm/connection.rb

```diff
@@ -8,36 +9,74 @@ class Connection < SimpleDelegator
     alias connection __getobj__
     alias connection= __setobj__
 
-    def initialize(connection:, default_log_prefix: nil, retry_options: {})
+    def initialize(connection:, default_log_prefix: nil, options: {}, retry_config: {})
       @default_log_prefix = default_log_prefix
-      @retry_options = retry_options || default_retry_config
+      @retry_options = retry_config || default_retry_config
       @sql_retry = Lhm::SqlRetry.new(
         connection,
-        retry_options,
+        options: retry_config,
+        reconnect_with_consistent_host: options[:reconnect_with_consistent_host] || false
       )
 
       # Creates delegation for the ActiveRecord Connection
       super(connection)
     end
 
-    def execute(query, retry_options = {})
-      exec_with_retries(:execute, query, retry_options)
+    def options=(options)
+      # If any other flags are added. Add the "processing" here
+      @sql_retry.reconnect_with_consistent_host = options[:reconnect_with_consistent_host] || false
```

Comment:
So if we do `connection.options = options`, it will only update a single member variable in `@sql_retry`? Why not just do `connections.reconnect_with_consistent_host = true/false` instead of using a options hash?

Reply:
I did because options could contain more than one values. I also added this so that there could be more feature flags in the future. However, if you don't see any new features coming in the future,  I can always make it more clear.